### PR TITLE
 #271 Renaming Spree to Solidus

### PR DESCRIPTION
**Pull request by JohnMorales**
This PR is for #271, While doing this I noticed things that will change which would need to be addressed:

1. This changes database tables in migrations, which could cause problems with existing databases.
1. All existing gems that reference the Spree namespace could potentially have issues.

That said, it was pretty straightforward.

### CODE REVIEW

**Comment by BenMorganIO** [comment]
For the gems that are going to reference `Spree` I would just do this somewhere:

```ruby
Spree = Solidus
```

**Comment by tvdeyen** [comment]
For gems using the `Spree` constant it's easy to migrate. Just like you did, replace `Spree` with `Solidus`. I wouldn't recommend to use something like `Spree = Solidus`. Expect strange, hard to debug errors. Rails eager and autoloading will not be happy with that. 

Otherwise 👍. Although I would love to see better commit messages. And a rebase that stashes commits that fixes one another. 

Regarding the migrations. One large table renaming migration is needed to make the upgrade possible.

File: LICENSE.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
-Spree License
+Solidus License
 =============
 
-Copyright © 2007-2014, Spree Commerce Inc. and other contributors.
+Copyright © 2007-2014, Solidus Commerce Inc. and other contributors.
```
Comment (mamhoff) [inline LICENSE.md]: Whoopsie! I don't think that Company exists. :)

File: LICENSE.md [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 All rights reserved.
 
 Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
 
 * Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
 * Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
-* Neither the name of Spree Commerce Inc. nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.
+* Neither the name of Solidus Commerce Inc. nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.
```
Comment (mamhoff) [inline LICENSE.md]: Solidus Commerce Inc. :)

File: README.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ Summary
 -------
 
 Solidus is a complete open source e-commerce solution built with Ruby on Rails. It
-is a fork of Spree.
+is a fork of Solidus.
```
Comment (mamhoff) [inline README.md]: Solidus is a fork of Spree.

File: README.md [hunk 4]
```diff
@@ -214,7 +214,7 @@ Name | Description | Badges |
...
+[Solidus Legacy Return Authorizations](https://github.com/solidusio/solidus_legacy_return_authorizations) | This is an extension for users migrating from legacy versions of Solidus (2.3.x and prior) which had a different representation of and handling for return authorizations. | [![Circle CI](https://circleci.com/gh/solidusio/solidus_legacy_return_authorizations/tree/master.svg?style=shield)](https://circleci.com/gh/solidusio/solidus_legacy_return_authorizations/tree/master) [![Gem](https://img.shields.io/gem/v/solidus_legacy_return_authorizations.svg)](https://rubygems.org/gems/solidus_legacy_return_authorizations)
```
Comment (mamhoff) [inline README.md]: ... legacy versions of Spree (not Solidus) in this context

File: api/LICENSE [hunk 5]
```diff
@@ -1,4 +1,4 @@
-Copyright (c) 2007-2014, Spree Commerce, Inc. and other contributors
+Copyright (c) 2007-2014, Solidus Commerce, Inc. and other contributors
```
Comment (mamhoff) [inline api/LICENSE]: Spree Commerce, Inc.

File: api/LICENSE [hunk 6]
```diff
@@ -9,7 +9,7 @@ are permitted provided that the following conditions are met:
     * Redistributions in binary form must reproduce the above copyright notice,
       this list of conditions and the following disclaimer in the documentation
       and/or other materials provided with the distribution.
-    * Neither the name Spree nor the names of its contributors may be used to
+    * Neither the name Solidus nor the names of its contributors may be used to
```
Comment (mamhoff) [inline api/LICENSE]: This should not be changed for the time being IMHO, and maybe there should be a bit of discussion about use of the name Solidus.

**Comment by tvdeyen** [comment]
Ah. Good catch @mamhoff  we must not by any means touch the current Spree license!! Maybe there is a way to dual licensing? Really don't know how to handle licenses of forks. But the originally one needs to stay untouched. This is what the license is all about, right?

**Comment by gmacdougall** [comment]
I think we're going to hold off on this change. There's lots of decisions that need to be made here, and it will break everyone's extensions, so we'll want to hold this for a major version if we want to do it at all.

**Comment by tvdeyen** [comment]
👍 for waiting until 2.0 (?)

**Comment by JohnMorales** [comment]
I agree that the LICENSE and README files would need to be hand reviewed. Possibly simply excluded. from a mass rename. 

@tvdeyen in reference to squashing the commits, I was concerned that making code changes and directory renames in a single commit would confuse git from keeping history across the renamed directory. That's why I did the directory rename separately, but I have no proof whether or not git can handle parent directory renames and file content changes without losing history.

That said this is a change that has to be done in a fell swoop, as it will get stale fairly quickly and really cannot co-exist with other branches. Therefore, I suggest we take the lessons learned, but the whole exercise should be redone when we're ready to actually pull the trigger.

**Comment by cbrunsdon** [comment]
@JohnMorales Aye, thanks for the attempt though. Hawthorn and I had a pretty serious conversation after this about what would be required to do this legitly. We might start sneaking in commits to help the transition in the log term. We're going to have to be really careful with migration/table names, any associations saved in the database (I.E. Spree::PaymentMethod stuff) and the like.

**Comment by JohnMorales** [comment]
yeah np, I understand why it makes sense to hold off, just keep in mind that it's almost never a good time to do something like this..

From someone that is considering using Solidus after building a prototype in Spree, I'd rather have a large change like this done before starting a new project.

**Pull request closed without merging**